lib/sycamore/sycamore/writers/file_writer.py

```diff
@@ -217,21 +220,48 @@ def get_filename_for_block(self, block: Block, task_index: int, block_index: int
         return f"block_{block_index}_{task_index}.jsonl"
 
 
-class _WritableFilePerRowDataSink(RowBasedFileDatasink):
+class _FileDataSink(Datasink):
     def __init__(
         self,
         path: str,
         filesystem: Optional[FileSystem] = None,
         filename_fn: Callable[[Document], str] = default_filename,
         doc_to_bytes_fn: Callable[[Document], bytes] = default_doc_to_bytes,
     ):
-        super().__init__(path, filesystem=filesystem, filename_provider=DocToRowFilenameProvider(filename_fn))
-
+        (paths, self._filesystem) = _resolve_paths_and_filesystem(path, filesystem)
+        self._root = paths[0]
+        print("ERIC DEBUG ROOT", self._root)
+        self._filename_fn = filename_fn
         self._doc_to_bytes_fn = doc_to_bytes_fn
 
-    def write_row_to_file(self, row: dict[str, Any], file: NativeFile):
-        binary = self._doc_to_bytes_fn(Document.from_row(row))
-        file.write(binary)
+    def write(self, blocks: Iterable[Block], ctx: TaskContext) -> Any:
+        for block in blocks:
+            b = BlockAccessor.for_block(block).to_arrow().to_pylist()
+            for _, row in enumerate(b):
+                doc = Document.from_row(row)
+                if isinstance(doc, MetadataDocument):
+                    continue
+                bytes = self._doc_to_bytes_fn(doc)
+                path = posixpath.join(self._root, self._filename_fn(doc))
+                with self._filesystem.open_output_stream(path) as file:
+                    file.write(bytes)
+
+
+# class _WritableFilePerRowDataSink(RowBasedFileDatasink):
+#     def __init__(
+#         self,
+#         path: str,
+#         filesystem: Optional[FileSystem] = None,
+#         filename_fn: Callable[[Document], str] = default_filename,
+#         doc_to_bytes_fn: Callable[[Document], bytes] = default_doc_to_bytes,
+#     ):
+#         super().__init__(path, filesystem=filesystem, filename_provider=DocToRowFilenameProvider(filename_fn))
+#
+#         self._doc_to_bytes_fn = doc_to_bytes_fn
+#
+#     def write_row_to_file(self, row: dict[str, Any], file: NativeFile):
+#         binary = self._doc_to_bytes_fn(Document.from_row(row))
+#         file.write(binary)
```

Comment:
here's the stuff that made me think this isn't compllete

Reply:
Removed.